relayer/src/listeners/eth.rs

```diff
@@ -91,49 +95,58 @@ impl EthListener {
 
         // Main Ethereum event loop.
         loop {
-            // Query for the next unknowns finalized block number, if not present we wait for it.
-            let next_finalized_block_number = get_next_finalized_block_number_eth(
-                eth_connection.clone(),
-                first_unprocessed_block_number,
-            )
-            .await;
-
-            // Don't query for more than `sync_step` blocks at one time.
-            let to_block = std::cmp::min(
-                next_finalized_block_number,
-                first_unprocessed_block_number + sync_step - 1,
-            );
-
-            log::info!(
-                "Processing events from blocks {} - {}",
-                first_unprocessed_block_number,
-                to_block
-            );
+            match emergency.load(Ordering::Relaxed) {
+                true => {
+                    trace!("Advisory is in an emergency state");
+                    sleep(Duration::from_millis(999)).await;
+                    continue;
+                }
+                false => {
+                    // Query for the next unknowns finalized block number, if not present we wait for it.
+                    let next_finalized_block_number = get_next_finalized_block_number_eth(
```

Comment:
Typically, relayer will wait here for several minutes, so it's most likely that if the emergency flag is set, the relayer will still process next batch of blocks (with the current implementation).

Similar situation with azero listener